bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/link/LinkBuilder.java

```diff
@@ -0,0 +1,61 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.commons.link;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Builds a link and sets link properties.
+ */
+public interface LinkBuilder {
+
+    /**
+     * Sets the property name to read the link URL from.
+     *
+     * @param name The property name
+     * @return {@link LinkBuilder}
+     */
+    @NotNull
+    LinkBuilder setLinkURLPropertyName(@Nullable String name);
```

Comment:
what's the usecase for passing null here? probably @NotNull is more appropriate?

Reply:
Addressed at https://github.com/adobe/aem-core-wcm-components/pull/2196/commits/91c774f82f21b3b8a5c3116ee3776f21783d4f59